src/ServiceControl.Config/Commands/StartServiceInMaintenanceModeCommand.cs

```diff
@@ -0,0 +1,67 @@
+﻿namespace ServiceControl.Config.Commands
+{
+    using System;
+    using System.Threading.Tasks;
+    using Caliburn.Micro;
+    using ServiceControl.Config.Events;
+    using ServiceControl.Config.Framework;
+    using ServiceControl.Config.Framework.Commands;
+    using ServiceControl.Config.UI.AdvanceOptions;
+    using ServiceControlInstaller.Engine.ReportCard;
+
+    class StartServiceInMaintenanceModeCommand : AwaitableAbstractCommand<AdvanceOptionsViewModel>
+    {
+        private readonly IWindowManagerEx windowManager;
+        private readonly IEventAggregator eventAggregator;
+
+        public StartServiceInMaintenanceModeCommand(IWindowManagerEx windowManager, IEventAggregator eventAggregator) : base(model => !(model.IsRunning && model.InMaintenanceMode))
+        {
+            this.windowManager = windowManager;
+            this.eventAggregator = eventAggregator;
+        }
+
+        public override async Task ExecuteAsync(AdvanceOptionsViewModel model)
+        {
+            model.ServiceControlInstance.Service.Refresh();
```

Comment:
This feels like a violation of https://en.wikipedia.org/wiki/Law_of_Demeter but I'm not sure how else you'd do it.